pkg/api/resource/helpers_test.go

```diff
@@ -17,9 +17,11 @@ limitations under the License.
 package resource
 
 import (
+	"fmt"
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
```

Comment:
I think we generally avoid using assert.

Reply:
Can you tell me the reason? I see that there are plenty of using assert in kubernetes now. :)

Reply:
Yes, it isnt enforced in any way.  See https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures.
In general, it is better to use the primitives that go provides rather than using a bunch of different libraries across the repo for testing, especially if you are essentially just checking equality.

Reply:
The guideline only mentioned that the failure message should include useful error messages. I don't think it says to not use an assert library.

I'll admit it. I've used the library a lot. It makes the tests more concise and readable and does not provide less information. Is there any issue in our repository discussing this?

Reply:
I dont believe so. I got the comment in a different code review. I guess its fine to use it here then

Reply:
OK, I will keep it.